src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -503,7 +503,7 @@ void SpatialPooler::compute(const UInt inputArray[], bool learn, UInt activeArra
 }
 
 
-void SpatialPooler::compute(SDR &input, bool learn, SDR &active) {
+void SpatialPooler::compute(const SDR &input, bool learn, SDR &active) {
```

Comment:
thanks to now possible const SDR, I can mark compute(const SDR input,...) to designate it's only input data.